src/main/java/thePackmaster/SpireAnniversary5Mod.java

```diff
@@ -707,24 +703,22 @@ public static AbstractCard getRandomCardFromPack(AbstractCardPack pack) {
         return validCards.get(AbstractDungeon.cardRandomRng.random(0, validCards.size() - 1)).makeCopy();
     }
 
-
-    public static ArrayList<AbstractCard> getCardsFromPacks(String pack, int count) {
+    public static ArrayList<AbstractCard> getCardsFromPacks(String pack, int count, Random rng) {
         ArrayList<String> quick = new ArrayList<>();
         quick.add(pack);
-        return getCardsFromPacks(quick, count);
+        return getCardsFromPacks(quick, count, rng);
     }
 
-    public static ArrayList<AbstractCard> getCardsFromPacks(ArrayList<String> packs, int count) {
+    public static ArrayList<AbstractCard> getCardsFromPacks(ArrayList<String> packs, int count, Random rng) {
```

Comment:
Passing the rng here (and in the method above) is the key change. There were also some formatting changes because the code was formatting in a really odd way that I wanted to improve.